crates/reth/evm/src/precompiles/schnorr.rs

```diff
@@ -0,0 +1,65 @@
+use std::array::TryFromSliceError;
+
+use revm::primitives::{PrecompileError, PrecompileOutput, PrecompileResult};
+use revm_primitives::Bytes;
+use strata_crypto::verify_schnorr_sig;
+use strata_primitives::buf::{Buf32, Buf64};
+use thiserror::Error;
+
+/// 32 Bytes: Public key
+/// 32 Bytes: Message hash
+/// 64 Bytes: Schnorr Signature
+struct SchnorrInput {
+    public_key: Buf32,
+    message_hash: Buf32,
+    signature: Buf64,
+}
+
+#[derive(Debug, Error)]
+enum SchnorrError {
+    #[error("Input not exactly 128 bytes")]
+    InvalidInput,
+
+    #[error("Slice conversion failed: {0}")]
+    FixedBytes(#[from] TryFromSliceError),
+}
+
+fn parse_schnorr_input(input: &Bytes) -> Result<SchnorrInput, SchnorrError> {
+    // validate if the length is 128 bytes or not
+    if input.len() != 128 {
+        return Err(SchnorrError::InvalidInput);
+    }
+
+    Ok(SchnorrInput {
+        public_key: Buf32::new(input[0..32].try_into().map_err(SchnorrError::FixedBytes)?),
+        message_hash: Buf32::new(input[32..64].try_into().map_err(SchnorrError::FixedBytes)?),
+        signature: Buf64::new(
+            input[64..128]
+                .try_into()
+                .map_err(SchnorrError::FixedBytes)?,
```

Comment:
These `FixedBytes` errors will never occur because the slice was sliced properly.  The variant can just be removed.  The only error case we have to consider here is if the input wasn't 128 bytes.